libs/optimizer/src/lib/fixtures/optimizer.fixtures.ts

```diff
@@ -0,0 +1,457 @@
+ */
+export function createFreeDistanceFixture(
+  from: LatLon,
+  to: LatLon,
+  nbSegments: number,
+  givenRules: ScoringRules,
+): OptimizerFixture {
+  const multiplier = getFreeDistanceMultiplier(givenRules);
+  const distance = getPreciseDistance(from, to) / 1000;
+  return {
+    givenRequest: {
+      track: createSegments(
+        { ...from, alt: 0, timeSec: 0 },
+        { ...to, alt: 0, timeSec: 60 },
+        START_TIME_SEC,
+        nbSegments,
+      ),
+    },
+    givenRules,
+    expectedResult: {
+      score: distance * multiplier,
+      lengthKm: distance,
+      multiplier,
+      circuit: CircuitType.OpenDistance,
+      optimal: true,
+    },
+  };
+}
+
+/**
+ *
+ * @returns a fixture for a free distance track with one intermediate bypass point and it's expected score
+ * @param from LatLon of the starting point of the free distance
+ * @param intermediate LatLon of the intermediate bypass point
+ * @param to LatLon of the ending point of the free distance
+ * @param nbSegments number of segments to create between each given points
+ * @param givenRules the ScoringRules for computing the score
+ */
+export function createFreeDistance1PointFixture(
+  from: LatLon,
+  intermediate: LatLon,
+  to: LatLon,
+  nbSegments: number,
+  givenRules: ScoringRules,
+): OptimizerFixture {
+  const distance = (getPreciseDistance(from, intermediate) + getPreciseDistance(intermediate, to)) / 1000;
+  const multiplier = getFreeDistanceMultiplier(givenRules);
+  return {
+    givenRequest: {
+      track: concatTracks(
+        createSegments(
+          { ...from, alt: 0, timeSec: 0 },
+          { ...intermediate, alt: 0, timeSec: 60 },
+          START_TIME_SEC,
+          nbSegments,
+        ),
+        createSegments(
+          { ...intermediate, alt: 0, timeSec: 60 },
+          { ...to, alt: 0, timeSec: 120 },
+          START_TIME_SEC,
+          nbSegments,
+        ),
+      ),
+    },
+    givenRules,
+    expectedResult: {
+      score: distance * multiplier,
+      lengthKm: distance,
+      multiplier,
+      circuit: CircuitType.OpenDistance,
+      optimal: true,
+    },
+  };
+}
+
+/**
+ *
+ * @returns a fixture for a free distance track with two intermediate bypass points and it's expected score
+ * @param from LatLon of the starting point of the free distance
+ * @param intermediate1 LatLon of the first intermediate bypass point
+ * @param intermediate2 LatLon of the second intermediate bypass point
+ * @param to LatLon of the ending point of the free distance
+ * @param nbSegments number of segments to create between each given points
+ * @param givenRules the ScoringRules for computing the score
+ */
+export function createFreeDistance2PointsFixture(
+  from: LatLon,
+  intermediate1: LatLon,
+  intermediate2: LatLon,
+  to: LatLon,
+  nbSegments: number,
+  givenRules: ScoringRules,
+): OptimizerFixture {
+  const distance =
+    (getPreciseDistance(from, intermediate1) +
```

Comment:
nit: formatting looks of. You can run `npx nx check` to format/lint/test before uploading a PR

Reply:
check performed. Formatting is ok now.
Added tip in CONTRIBUTING.md